### PR TITLE
Bbmap index

### PR DESCRIPTION
**Pull request by erikrikarddaniel**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

Adds a module that creates a `BBMap` index.

I really don't need the `meta` map, but it didn't lint when I removed it.

### CODE REVIEW

File: modules/bbmap/index/main.nf [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+def VERSION = '38.92'       // There seems to be no way to get the version out of the bbmap.sh script, only a line with "Last modified" and a date
```
Comment (jfy133) [inline modules/bbmap/index/main.nf:7]: See here how to do it: https://github.com/nf-core/modules/blob/master/modules/bbmap/bbduk/main.nf

File: modules/bbmap/index/main.nf [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        container "https://depot.galaxyproject.org/singularity/bbmap:38.92--he522d1c_0"
+    } else {
+        container "quay.io/biocontainers/bbmap:38.92--he522d1c_0"
+    }
+
+    input:
+    tuple val(meta), path(fasta)
+
+    output:
+    tuple val(meta), path('ref')  , emit: db
```
Comment (jfy133) [inline modules/bbmap/index/main.nf:27]:
```suggestion
    tuple val(meta), path('ref')  , emit: db
```

Maybe just `index` for the channel? To keep consistency with the module name?

**Review by jfy133 — CHANGES_REQUESTED** [review]
Couple of suggestions but overal LGTM

File: modules/bbmap/index/main.nf (hunk 1 above)
Comment (erikrikarddaniel) [reply modules/bbmap/index/main.nf:7]: Thanks, didn't know that.

File: modules/bbmap/index/main.nf (hunk 2 above)
Comment (erikrikarddaniel) [reply modules/bbmap/index/main.nf:27]: I can't see any difference, did you mean `emit: index`?

Comment (jfy133) [reply modules/bbmap/index/main.nf:25]: Yes sorry. Early morning reviews not the best idea :laughing:

Comment (erikrikarddaniel) [reply modules/bbmap/index/main.nf]: :-) I just pushed a separate commit with this.

**Review by jfy133 — APPROVED** [review]
Once tests pass

**Pull request merged**